lib/src/rules/avoid_type_to_string.dart

```diff
@@ -74,12 +74,7 @@ class AvoidTypeToString extends LintRule implements NodeLintRule {
 
     // These nodes delegate to general visitArgumentList,
     // since SimpleAstVisitor only calls visits for concrete node subtypes.
-    // TODO: replace these with addArgumentList(...) once it's added to the registry API.
-    // Context: https://github.com/dart-lang/linter/pull/2209#discussion_r465196745
-    registry.addFunctionExpressionInvocation(this, visitor);
-    registry.addInstanceCreationExpression(this, visitor);
-    registry.addRedirectingConstructorInvocation(this, visitor);
-    registry.addSuperConstructorInvocation(this, visitor);
+    registry.addArgumentList(this, visitor);
```

Comment:
The methods that are no longer being registered should be removed from the visitor because they are no longer being invoked and the comment on lines 75-76 should be removed as it no longer applies.